frontend/crd/src/domain/Task.ts

```diff
@@ -1,20 +1,23 @@
 export interface TaskJSON {
-    description: string;
-    needsArtifact: Boolean;
-    id: string;
-    isRequired: Boolean;
+  description: string;
+  needsArtifact: boolean;
+  id: string;
+  isRequired: boolean;
 }
 
 export class Task {
-    constructor(json: TaskJSON) {
-        this.description = json.description;
-        this.needsArtifact = json.needsArtifact;
-        this.id = json.id;
-        this.isRequired = json.isRequired;
-    }
+  constructor(json: TaskJSON) {
+    this.description = json.description;
+    this.needsArtifact = json.needsArtifact;
+    this.id = json.id;
+    this.isRequired = json.isRequired;
+    // TODO this should be checked through the submission object or provided by the backend later
+    this.isComplete = true;
```

Comment:
would there also be a field for the submitted artifact or the submission in general?

Reply:
yes, but we haven't done the submission object yet. I think it would kind of be backward to work on handling/displaying submissions before we even have a way to create a submission you know?

Reply:
I was thinking of at least having the field there, even if it's just a placeholder for now

Reply:
sure I'll add a submission field with type any for now.